### PR TITLE
Migrate compiler config to Config

### PR DESCRIPTION
**Pull request by oulgen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143152
* #143229

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/compiler/config.py [hunk 1]
```diff
@@ -28,7 +29,7 @@
 # FB-internal note: you do NOT have to specify this explicitly specify this if
 # you run on MAST, we will automatically default this to
 # mast:MAST_JOB_NAME:MAST_JOB_VERSION.
-job_id: Optional[str] = os.environ.get("TORCH_COMPILE_JOB_ID", None)
+job_id: Optional[str] = Config(env_name_default="TORCH_COMPILE_JOB_ID", default=None)
```
Comment (bobrenjc93) [inline torch/compiler/config.py:32]: Hmmm Config seems to only support Bool... so we either should have expanded Config to support strings or not done this refactor.

```
torch.compiler.config.job_id
True
```

**Pull request closed without merging**